chacra/tests/async/test_recurring.py

```diff
@@ -37,22 +37,142 @@ def test_gets_rid_of_old_repos(self, session, fake, monkeypatch):
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'ref': {'next': {'keep_minimum': 0}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
+
+    def test_get_rid_of_old_repos_by_flavor_configured_in_days(self, session, fake, monkeypatch):
+        Repo(
+            project=Project('nfs-ganesha'),
+            ref='next',
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'flavor': {'ceph_master': {'days': 0}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
+
+    def test_get_rid_of_old_repos_by_flavor_configured_with_offset(self, session, fake, monkeypatch):
+        Repo(
+            project=Project('nfs-ganesha'),
+            ref='next',
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'flavor': {'ceph_master': {'keep_minimum': 0}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
+
+    def test_does_not_get_rid_of_old_repos_by_flavor_and_ref(self, session, fake, monkeypatch):
+        conf.purge_rotation = {'ceph': {'flavor': {'default': {'keep_minimum': 1}}, 'ref': {'firefly': {'days': 70}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 1
+
+    def test_get_rid_of_old_repos_by_flavor_with_ref(self, session, fake, monkeypatch):
+        Repo(
+            project=Project('nfs-ganesha'),
+            ref='next',
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'flavor': {'ceph_master': {'keep_minimum': 0}}, 'ref': {'next': {'days': 70}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
```

Comment:
AhI envisioned any condition in the purge rotation that says the repo would be purged signals the repo to be removed. Should I change it around so that all conditions have to agree to a purge? Also is this even a situation worth testing?